src/libproxychains.c

```diff
@@ -336,7 +336,7 @@ static int is_v4inv6(const struct in6_addr *a) {
 	return a->s6_addr32[0] == 0 && a->s6_addr32[1] == 0 &&
 	       a->s6_addr16[4] == 0 && a->s6_addr16[5] == 0xffff;
 }
-int connect(int sock, const struct sockaddr *addr, unsigned int len) {
+int connect(int sock, const struct sockaddr *addr, socklen_t len) {
```

Comment:
this change can open a new can of worms, since the build will fail on every platform where the function prototype is only even cosmetically different from what the libc headers declare.
that means that this change will eventually break the build on musl libc, uclibc, newlib, freebsd, etc and does not even fix a warning.